perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -4741,8 +4753,39 @@ public void removeGroupUnion(PerunSession sess, Group resultGroup, Group operand
 
 		// recalculates statuses of members in result group
 		for (Member member : getGroupMembers(sess, resultGroup)) {
-			recalculateMemberGroupStatusRecursively(sess, member, resultGroup);
+			recalculateMemberGroupStatusRecursively(sess, member, resultGroup, previousStatuses);
+		}
+	}
+
+	/**
+	 * Returns previous total member-group statuses of members in all result groups of the given group.
+	 *
+	 * @param sess perun session
+	 * @param group group
+	 * @param members members
+	 * @param previousStatuses previous statuses, empty in at the beginning
+	 * @return previous statuses of members in all result groups of the given group
+	 */
+	private Map<Pair<Group, Member>, MemberGroupStatus> getPreviousStatuses(PerunSession sess, Group group, List<Member> members, Map<Pair<Group, Member>, MemberGroupStatus> previousStatuses) {
```

Comment:
It would be nice to create an overloaded version of this method, so you don't have to pass an empty map every time you want to call it: 
```Java
private Map<Pair<Group, Member>, MemberGroupStatus> getPreviousStatuses(PerunSession sess, Group group, List<Member> members, Map<Pair<Group, Member>, MemberGroupStatus> previousStatuses) {
    return getPreviousStatuses(sess, group, members, new HashMap<>);
}
```

Reply:
Added.